examples/tn_simulation.py

```diff
@@ -0,0 +1,157 @@
+print("Expectation value for Z^n: ", exp_val)
+
+
+# If we want to find the solution for our initial max-cut problem,
+# then we must deploy a classical minimizer too for an apropriate cost function.
+# Create a cost function using the elements of graphix, which were already discussed above.
+def cost(params, n, ham, quantum_iter, slice_index, opt=None):
+    circuit = Circuit(n)
+    gamma = params[:slice_index]
+    beta = params[slice_index:]
+    ansatz(circuit, n, gamma, beta, quantum_iter)
+
+    pattern = circuit.transpile(opt=True).pattern
+    pattern.standardize()
+    pattern.shift_signals()
+    pattern.perform_pauli_measurements(use_rustworkx=True)
+    mbqc_tn = pattern.simulate_pattern(backend="tensornetwork", graph_prep="parallel")
+    exp_val = 0
+    for op in ham:
+        exp_val += np.real(mbqc_tn.expectation_value(op, range(n), optimize=opt))
+    return exp_val
+
+
+# We want to find the ground state energy for the Hamiltonian = Sum(Z_k) + Sum(Z_iZ_j) with i,j running over the edges.
+ham = [reduce(np.kron, [pauli_z] * n)]
+for i in range(1, n):
+    op = [identity] * n
+    op[0] = pauli_z
+    op[i] = pauli_z
+    op = reduce(np.kron, op)
+    ham.append(op)
+
+
+# Use yet again another optimizer for path contraction.
+class MyOptimizer(oe.paths.PathOptimizer):
+    def __call__(self, inputs, output, size_dict, memory_limit=None):
+        return [(0, 1)] * (len(inputs) - 1)
+
+
+opt = MyOptimizer()
+# Define initial parameters, which will be optimized through running the algorithm.
+params = 2 * np.pi * np.random.rand(len(gamma) + len(beta))
+# Run the classical optimizer and simulate the quantum circuit with TN backend.
+res = minimize(cost, params, args=(n, ham, iterations, len(gamma), opt), method="COBYLA")
+print(res.message)
+
+# Finally, run the circuit once again with the optimized parameters.
+circuit = Circuit(n)
+ansatz(circuit, n, res.x[: len(gamma)], res.x[len(gamma) :], iterations)
+pattern = circuit.transpile(opt=True).pattern
+pattern.standardize()
+pattern.shift_signals()
+mbqc_tn = pattern.simulate_pattern(backend="tensornetwork", graph_prep="parallel")
+
+# Let's use the defined optimizer and find the most probable basis states.
+max_prob = 0
+most_prob_state = 0
+bars = []
+for i in range(0, 2**n):
+    value = mbqc_tn.get_basis_amplitude(i)
+    bars.append(value)
+
+# Plot the output.
+fig, ax = plt.subplots(figsize=(10, 5))
+ax.bar(range(0, 2**n), bars, color="maroon", width=0.2)
+ax.set_xticks(range(0, 2**n))
+ax.set_xlabel("States")
+ax.set_ylabel("Probabilites")
+ax.set_title("Measurement probabilities using the optimized parameters")
+plt.show()
+# As we can see the most probable are 15 and 16 (|11110> and |00001> because of bit ordering),
+# which mean that splitting the graph so that node number 0 is in one set,
+# and all other nodes in the other solves the max cut problem.
+# This result is what we would expect from this star-like graph.
```

Comment:
looks good, can you do a quick visualization of max-cutting that's done here?

Reply:
Of course, I will include additional visualization at the end